WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -513,7 +513,7 @@ public async Task<ConnectionConfirmationResponse> ConfirmConnectionAsync(Connect
 			}
 		}
 
-		public async Task RegisterOutputAsync(OutputRegistrationRequest request, CancellationToken cancellationToken)
+		public async Task<OutputRegistrationResponse> RegisterOutputAsync(OutputRegistrationRequest request, CancellationToken cancellationToken)
```

Comment:
mainly out of curiosity, is something like `Task<void>` possible?

Reply:
just use `Task` :)

Reply:
I was assuming that some dummy type was needed, hence this new unit type called `OutputRegistrationResponse`, but come to think of it i don't actually understand why that's necessary

Reply:
I think we need to add `Unit` type after all. It simplifies things a lot. Otherwise we have to duplicate code in many places.

Reply:
So basically there are two choices:

* add `Unit` type as Lucas says, or
* add `public async Task GetCachedResponseAsync<TRequest>(TRequest request, ProcessRequestDelegateAsync<TRequest> action, CancellationToken cancellationToken = default)`. This seems like it would involve more code, more testing, etc.

`OutputRegistrationResponse` is confusing because it's just a unit type really. So unit type seems like a good option here.

> mainly out of curiosity, is something like `Task<void>` possible?

No, AFAIK.

Reply:
> That said, `Unit` or even "unit type" generally only means anything to people with a functional programming background, so maybe it's not the clearest name.

Yes, those who have not encountered FP may be confused. `EmptyResponse` sounds good to me.

Reply:
We are already using Unit type in the UI and I do not agree with the idea that what we can understand with no problem can be confused for others and that we have to create fool-proof names for those.

Reply:
I see. I didn't know that Unit type has already been introduced.